projects/client/RabbitMQ.Client/src/client/api/ConnectionFactory.cs

```diff
@@ -240,14 +240,33 @@ public AmqpTcpEndpoint Endpoint
         public String Uri
         {
             set { SetUri(new Uri(value, UriKind.Absolute)); }
+            get {
+                var protocol="amqp";
+                if(Ssl!=null&& Ssl.Enabled) {
+                    protocol="amqps";
+                }
+                var uriString=protocol+"://"+HostName+":"+Port+"/";
+                if(VirtualHost!=null&&!VirtualHost.Equals("")) {
+                    uriString=uriString+VirtualHost;
+                }
+                return uriString;
+            }
         }
 
+        private IList<Uri> _uris;
         /// <summary>
-        /// Set connection parameters using the amqp or amqps scheme.
+        /// Set a list of conection parameters (only for autorecovering connections)
         /// </summary>
-        public Uri uri
```

Comment:
You are breaking public API here. Please restore the removed method, it probably can delegate to `Uris` with a single element list.